WI #647: Adds `Amortization Class` column (#747)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: supporting_content/columns/amortizationcategory.md

# Column: Amortization Category

> Objective: we need a simple method to calculate savings that doesn't accidentally double-count commitment discount purchase costs.

Reviewer: ```suggestion
Objective: we need a simple method to calculate savings that doesn't accidentally double-count amortized purchase costs.
```

Reviewer: Monthly fees can still be amortized. Amortization [as I understand it] is the process of taking any charge and splitting it out by the time granularity of the dataset. So if I amortize a charge that has a 7-day applicability period, then I would see 7 amortized rows in a daily dataset and 168 amortized rows in an hourly dataset. Conversely, if I have a monthly dataset, then there'd only be 1 amortized row. All of these are on top of the original/principal/source charge.

Author: Agree, per above.

Objective: we need a simple method to identify which line items are and are not eligible for

> commitment discounts, and which actually have commitment discounts applied to them.

Reviewer: This feels separate from amortization. I do think this is something we need, but it needs to account for usage and spend commitment discounts. I think that needs to be a separate column (or 2).

## Example provider mappings

Current column mappings found in available data sets:

| Provider | Data set                 | Column                    |
|----------|--------------------------|---------------------------|
| AWS      | CUR                      | lineItem/lineItemType     |
| Google Cloud | BigQuery Billing Export | ??N/A because no upfront payments??  |

> | Microsoft | Cost details             | ?? Specific SKUs that designate upfront payments |

Reviewer: ```suggestion
| Microsoft    | Cost details             | ChargeType indicates purchases and PricingModel indicates commitment discounts. There is no column that identifies when and how amortization applies. |
```

VM usage | ... | ... | ... | ... | ...
VM usage | Dec 31 | $2.00 | $1.80 | $0.00 | $1.00
Sum of 365 days | Jan 1-Dec 31 | $1,095.00 | $1,022.00 | $365 | $365

Example #1

If you SUM the ListCost or ContractedCost columns as is, you'll be double counting the original "Reservation" charge and overcounting the savings present in the e.g. EffectiveCost column. We need an attribute that allows us to selectively exclude upfront payments from such calculations, so that we are only SUMming the usage charges that benefit from the original upfront payment (or "unused commitment" line items).

`SUMIF(ListCost, AmortizationCategory != "Commitment Purchase")` would give you the correct total of $730.00, which is the sum of the usage charges that benefit from the original upfront payment.

Example #2

For calculating something like "commitment coverage", you can use the "Eligible - Covered" and "Eligible - Not Covered" values. For example, if you want to know the balance of covered to "not covered, but potentially coverable" you could use:

```
SELECT
    SUM(ListCost) as TotalCost,
    AmortizationCategory
...
GROUP BY AmortizationCategory
```

## Discussion / Scratch space:

> 

Reviewer: ```suggestion

There are 4 known scenarios for double-counting:
1. Amortization
2. Burndown of a previous purchase with no charge for underutilized purchases
3. Burndown of a promise where no money is charged with a potential "unused" charge at the end of the period
4. Marketplace rows in a cloud provider's dataset and separately in a SaaS provider's dataset being combined by the practitioner

Amortization requirements:
* Must be able to identify (and filter out) the original/source charge.
  * This charge has a BilledCost of some positive value and an EffectiveCost of 0.
  * This charge is the one that will get filtered out when calculating savings.
  * This is typically a purchase but may not always be (e.g., monthly aggregated usage row could be broken down into daily/hourly usage rows).
  * In finance, the original charge is generally referred to as the "principal".
* Must be able to identify (and filter down to) the target or "effective" charges where the principal was amortized out over time.
* All amortized charges should include rows for both the source and target charges to ensure the filtering works as intended.
```
